src/Microsoft.TemplateEngine.Abstractions/IGenerator.cs

```diff
@@ -57,6 +58,29 @@ public interface IGenerator : IIdentifiedComponent
         /// <returns><see cref="IParameterSet"/> with parameters available in <paramref name="template"/>.</returns>
         IParameterSet GetParametersForTemplate(IEngineEnvironmentSettings environmentSettings, ITemplate template);
 
+        /// <summary>
+        /// Evaluates possible conditions defined on parameters within the given set of parameters.
+        /// Performs the changes on the given set of parameters as needed and returns the summary of changes performed.
+        /// </summary>
+        /// <param name="logger"></param>
+        /// <param name="parameters">Parameters to be evaluated and updated.</param>
+        /// <param name="template"></param>
+        /// <returns></returns>
+        ParametersConditionsEvaluationResult EvaluateConditionalParameters(ILogger logger, IParameterSet parameters, ITemplate template);
+
+        /// <summary>
+        /// Applies result of external conditions evaluation as if the evaluation was performed via <see cref="EvaluateConditionalParameters"/>.
+        /// This means removing the disabled parameters from values and templates collections and altering the requirement of parameters with
+        ///  conditional requirement.
+        /// </summary>
+        /// <param name="evaluationResult"></param>
+        /// <param name="parameterSet"></param>
+        /// <param name="template"></param>
+        void ApplyExternalEvaluationOfConditionalParameters(
```

Comment:
Should this method be on `IParameterSet` instead?

Reply:
Probably yes. Then in needs to be on a template as well (as e.g. for replacements we use the symbols, not parameters - so those need to be removed as well: https://github.com/dotnet/templating/blob/main/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectConfig.cs#L776)